src/field/DormandPrinceStepper.hh

```diff
@@ -0,0 +1,206 @@
+    constexpr R a32 = -56 / R(15);
+    constexpr R a33 = 32 / R(9);
+
+    constexpr R a41 = 19372 / R(6561);
+    constexpr R a42 = -25360 / R(2187);
+    constexpr R a43 = 64448 / R(6561);
+    constexpr R a44 = -212 / R(729);
+
+    constexpr R a51 = 9017 / R(3168);
+    constexpr R a52 = -355 / R(33);
+    constexpr R a53 = 46732 / R(5247);
+    constexpr R a54 = 49 / R(176);
+    constexpr R a55 = -5103 / R(18656);
+
+    constexpr R a61 = 35 / R(384);
+    constexpr R a63 = 500 / R(1113);
+    constexpr R a64 = 125 / R(192);
+    constexpr R a65 = -2187 / R(6784);
+    constexpr R a66 = 11 / R(84);
+
+    constexpr R d71 = a61 - 5179 / R(57600);
+    constexpr R d73 = a63 - 7571 / R(16695);
+    constexpr R d74 = a64 - 393 / R(640);
+    constexpr R d75 = a65 + 92097 / R(339200);
+    constexpr R d76 = a66 - 187 / R(2100);
+    constexpr R d77 = -1 / R(40);
+
+    // Coefficients for the mid point calculation by Shampine
+    constexpr R c71 = 6025192743 / R(30085553152);
+    constexpr R c73 = 51252292925 / R(65400821598);
+    constexpr R c74 = -2691868925 / R(45128329728);
+    constexpr R c75 = 187940372067 / R(1594534317056);
+    constexpr R c76 = -1776094331 / R(19743644256);
+    constexpr R c77 = 11237099 / R(235043384);
+
+    Result result;
+
+    // First step
+    OdeState k1    = equation_(beg_state);
+    OdeState state = beg_state;
+    axpy(a11 * step, k1, &state);
+
+    // Second step
+    OdeState k2 = equation_(state);
+    state       = beg_state;
+    axpy(a21 * step, k1, &state);
+    axpy(a22 * step, k2, &state);
+
+    // Third step
+    OdeState k3 = equation_(state);
+    state       = beg_state;
+    axpy(a31 * step, k1, &state);
+    axpy(a32 * step, k2, &state);
+    axpy(a33 * step, k3, &state);
+
+    // Fourth step
+    OdeState k4 = equation_(state);
+    state       = beg_state;
+    axpy(a41 * step, k1, &state);
+    axpy(a42 * step, k2, &state);
+    axpy(a43 * step, k3, &state);
+    axpy(a44 * step, k4, &state);
+
+    // Fifth step
+    OdeState k5 = equation_(state);
+    state       = beg_state;
+    axpy(a51 * step, k1, &state);
+    axpy(a52 * step, k2, &state);
+    axpy(a53 * step, k3, &state);
+    axpy(a54 * step, k4, &state);
+    axpy(a55 * step, k5, &state);
+
+    // Sixth step
+    OdeState k6      = equation_(state);
+    result.end_state = beg_state;
+    axpy(a61 * step, k1, &result.end_state);
+    axpy(a63 * step, k3, &result.end_state);
+    axpy(a64 * step, k4, &result.end_state);
+    axpy(a65 * step, k5, &result.end_state);
+    axpy(a66 * step, k6, &result.end_state);
+
+    // Seventh step: the final step
+    OdeState k7 = equation_(result.end_state);
+
+    // The error estimate
+    axpy(d71 * step, k1, &result.err_state);
+    axpy(d73 * step, k3, &result.err_state);
+    axpy(d74 * step, k4, &result.err_state);
+    axpy(d75 * step, k5, &result.err_state);
+    axpy(d76 * step, k6, &result.err_state);
+    axpy(d77 * step, k7, &result.err_state);
```

Comment:
It might be more error-proof (but less performant?) to write this sequence as a triangular loop, something like (this is incomplete):
```c++
Array<OdeState, 7> k;
static const real_type coeff[] = {a11, a21, a22, a31, ...};
int ij = 0;
for (int i : range(7))
{
    k[i] = equation_(state);
    for (int j : range(i + 1))
    {
        axpy(coeff[ij++] * step, k[j], result.end_state);
    }
}
```

Reply:
Actually I tried something similar, but found that performance was significantly worse (25ms vs 40ms in CPU in the stepper test), so I decided to go for this brutal way - also note that a62 and d62 do not exist (well, we can set them to 0 and do dummy calculations in the loop though).  I will revisit this type of simplification (which I always love to do) later, but let's put this first implementation in as it is for now.  Thanks for the suggestion anyway.

Reply:
Fascinating... glad you tried! I didn't notice the lack of a62/d62... it would take some time for me to notice *any* subtle change in there 😬  Definitely leave this in for now.